src/test/java/org/opentripplanner/netex/loader/util/HierarchicalMapTest.java

```diff
@@ -0,0 +1,144 @@
+package org.opentripplanner.netex.loader.util;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Collection;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.opentripplanner.netex.loader.util.E.EASTWOOD;
+import static org.opentripplanner.netex.loader.util.E.REAGAN;
+import static org.opentripplanner.netex.loader.util.E.SCHWARZENEGGER;
+import static org.opentripplanner.netex.loader.util.SetSupport.listOf;
+import static org.opentripplanner.netex.loader.util.SetSupport.sort;
+
+/**
+ * To test the hierarchical map we construct a hierarchy of:
+ * <pre>
+ *     Country(root) -> State -> County(leaf)
+ * </pre>
+ * We add <em>president</em> <b>Regan</b> to <b>country</b>, a <em>governor</em>
+ * <b>Schwarzenegger</b> to <b>state</b> and <em>mayor</em> <b>Eastwood</b> to <b>county</b>.
+ * <p/>
+ * We also add an <em>actor</em> for each level, but this time <b>Eastwood</b> is added to the
+ * <b>country</b> level (clearly the best actor ;-). <b>Schwarzenegger</b> is
+ * added to the <b>state</b> level, and <b>Reagan</b> to <b>county</b>.
+ * <pre>
+ * Level\Role | President | Governor       | Mayor    | Actor
+ * -----------|-----------|----------------|----------|----------------
+ *    Country | Reagan    |                |          | Eastwood
+ *      State |           | Schwarzenegger |          | Schwarzenegger
+ *     County |           |                | Eastwood | Reagan
+ * </pre>
+ * Now we use this test setup to test the {@link HierarchicalMap} and
+ * {@link AbstractHierarchicalMap} class.
+ */
+public class HierarchicalMapTest {
+    private static final String PRESIDENT = "president";
+    private static final String GOVERNOR = "governor";
+    private static final String MAYOR = "mayor";
+    private static final String ACTOR = "actor";
+
+    private final HierarchicalMap<String, E> country = new HierarchicalMap<>();
+    private final HierarchicalMap<String, E> state = new HierarchicalMap<>(country);
+    private final HierarchicalMap<String, E> county = new HierarchicalMap<>(state);
+
+
+    @Before
+    public void setup() {
+        country.add(PRESIDENT, REAGAN);
+        state.add(GOVERNOR, SCHWARZENEGGER);
+        county.add(MAYOR, EASTWOOD);
```

Comment:
It looks like Eastwood was briefly mayor of Carmel-by-the-sea, California, which is a city rather than a county. Recommend replacing changing `county` to `city = new HierarchicalMap(state)`. 😉